### PR TITLE
Add run_bolt_plan helper

### PR DESCRIPTION
**Pull request by RandomNoun7**
Add run_bolt_plan helper

This change adds the ability for Litmus to run plans.

Plans are different from tasks in two major ways:

1. The target to run a plan against is a parameter to the plan, not to
Bolt. This means that target host is going to be a parameter to the plan
that the plan author has named whatever they like. As such the method
signature for the underlying `run_plan` helper in Boltspec has no target
param. As such
when calling a plan using this helper, it is up to the caller to know
the name of the parameter in the plan that will accept the target URI
and include the parameter and it's value in the `params` hash.

2. A plan is not a single executable that is copied to a target host
and run. As such there are no standard `exit_code`, `stdout` or `stderr`
nodes in the return we get from the Boltspec helper. We get only
`status` and `value` (A string array) if success, or `status` and
`value` (A complex object) if failure.

For backward compatibility and for ease of testing this new helper
creates a fake exit code, and it creates `stdout` and `stderr` keys in
the return object. The exit code is simply `0` if the plan succeeded and
`1` if it failed. The `stdout` key a string which is the `value` key of
the real result joined by line breaks. `Stderr` is is simply the `msg`
property of the `value` key if the result was a failure. If the result
was a success then `stderr` is nil.

Since `target_node_name` is not passed as a parameter to the underlying
Boltspec helper, it's only purpose here, and the reason it still exists
in this new function, is solely as a check to ensure that the target you
are about to pass to the plan exists in the inventory file you have
provided.

### CODE REVIEW

**Comment by RandomNoun7** [comment]
Let me know if you would like this change. If you do I'll go ahead and write the tests. If not I'll figure something else out to get this done.

Running plans would be very useful for me. My current use case is I want to execute a plan in the [peadm](https://github.com/puppetlabs/puppetlabs-peadm) module that installs Puppet Enterprise. Using this module is a best practice for installing PE as it is maintained by the people at Puppet most interested in ensuring it functions properly and because it is far more flexible than anything I could write.

The inability for Litmus to run plans makes using this module impossible.

File: lib/puppet_litmus/puppet_helpers.rb [hunk 1]
```diff
@@ -303,6 +303,64 @@ def run_bolt_task(task_name, params = {}, opts = {})
...
+      span.add_field('litmus.node_name', target_node_name)
+      add_platform_field(inventory_hash, target_node_name)
+
+      bolt_result = run_plan(plan_name, params, config: config_data, inventory: inventory_hash)
+      # Plans don't provide stdin and stdout streams. They only have a result.
+      # They don't provide an exit code either, just a 'success' value.
+      result_obj = {
+        status: nil,
+        result: nil,
+      }
```
Comment (tphoney) [inline lib/puppet_litmus/puppet_helpers.rb:341]:
```
result_obj = {
        exit_code: 0,
        stdout: nil,
        stderr: nil,
        result: bolt_result.first['value'],
      }
```
The reason for this is to allow backwards compatibility for beaker style tests. 
Granted we are adding something that beaker never had, but people still write tests where they check for status code, even though there is no status code. they just want to know if it failed with a .zero? check
You should be able to copy from https://github.com/puppetlabs/puppet_litmus/blob/b4f659013855d5900f2481bd2cc9fc2f56b7af2f/lib/puppet_litmus/puppet_helpers.rb#L245

Comment (RandomNoun7) [reply lib/puppet_litmus/puppet_helpers.rb:341]: Yeah I can add those. I'll just a comment in the code that they aren't real. Just placeholders as you say for compatibility.

Comment (RandomNoun7) [reply lib/puppet_litmus/puppet_helpers.rb:341]: Can I get away with just adding the exit code and not sdtout and stderr?

Comment (RandomNoun7) [reply lib/puppet_litmus/puppet_helpers.rb:341]: Don't mind me. I decided to add those in too.

**Comment by RandomNoun7** [comment]
There seem to be a good number of rubocop violations that aren't caused by my PR. I don't know if you want to wait until those are resolved and tests pass before getting this in or not.

**Comment by RandomNoun7** [comment]
Ping

**Pull request closed without merging**